File: app/src/main/java/com/emikhalets/datesdb/viewmodels/DateEditViewModel.java
package com.emikhalets.datesdb.viewmodels;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;

import com.emikhalets.datesdb.data.AppDatabase;
import com.emikhalets.datesdb.data.AppRepository;
import com.emikhalets.datesdb.data.DateItem;

public class DateEditViewModel extends AndroidViewModel {

    private AppRepository repository;

    public DateEditViewModel(Application application) {
        super(application);

        repository = new AppRepository(AppDatabase.getInstance(application).getDatesDao());
    }

    public void insert(String name, String date, String type) {
        DateItem dateitem = new DateItem(name, date, type);
        repository.insert(dateitem);
    }
}

File: app/src/main/java/com/emikhalets/datesdb/viewmodels/DateEditViewModel.kt
package com.emikhalets.datesdb.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.emikhalets.datesdb.data.AppRepository.Companion.getInstance

class DateEditViewModel(application: Application?) : AndroidViewModel(application!!) {
    private val repository: AppRepository?

    init {
        repository = getInstance(application!!)
    }
}